src/main/java/org/quiltmc/loader/api/config/Config.java

```diff
@@ -0,0 +1,423 @@
+
+	/**
+	 * Save this config and all its values to disk
+	 */
+	void serialize();
+
+	/**
+	 * Returns all values held by this config file
+	 *
+	 * <p>For all nodes, including section nodes, see {@link #nodes}
+	 *
+	 * @return all values held by this config file
+	 */
+	Iterable<TrackedValue<?>> values();
+
+	/**
+	 * @param key an iterable of key components to access
+	 * @return the value contained by this config class with key components
+	 */
+	TrackedValue<?> getValue(Iterable<String> key);
+
+	/**
+	 * @return all nodes of the value tree represented by this config file, including section nodes
+	 */
+	Iterable<ValueTreeNode> nodes();
+
+	/**
+	 * Creates and registers a config with fields derived from the fields of the passed class
+	 *
+	 * <p>The passed class should have the following characteristics:
+	 * <ul>
+	 *     <li>Has a public no-argument constructor</li>
+	 *     <li>Each non-static field should be final, not null, and be one of the following types:</li>
+	 *     <ul>
+	 *         <li>A basic type (int, long, float, double, boolean, String, or enum)</li>
+	 *         <li>A complex type (a {@link ValueList} or {@link ValueMap} of basic or complex types)</li>
+	 *         <li>An object whose class follows these rules</li>
+	 *     </ul>
+	 * </ul>
+	 *
+	 * @param modId the mod owning the resulting config file
+	 * @param id the configs id
+	 * @param path additional path elements to include as part of this configs file, e.g.
+	 *             if the path is empty, the config file might be ".minecraft/config/example_mod/id.json5"
+	 *             if the path is "client/gui", the config file might be ".minecraft/config/example_mod/client/gui/id.json5"
+	 * @param creators any number of {@link Creator}s that can be used to configure the resulting config
+	 */
+	static Config create(String modId, String id, Path path, Creator... creators) {
+		return ConfigImpl.create(modId, id, path, creators);
+	}
+
+	/**
+	 * Creates and registers a config
+	 *
+	 * @param modId the mod owning the resulting config file
+	 * @param id the configs id
+	 * @param creators any number of {@link Creator}s that can be used to configure the resulting config
+	 */
+	static Config create(String modId, String id, Creator... creators) {
+		return create(modId, id, Paths.get(""), creators);
+	}
+
+	/**
+	 * Creates and registers a config with fields derived from the fields of the passed class
+	 *
+	 * <p>The passed class should have the following characteristics:
+	 * <ul>
+	 *     <li>Has a public no-argument constructor</li>
+	 *     <li>Each non-static field should be final, not null, and be one of the following types:</li>
+	 *     <ul>
+	 *         <li>A basic type (int, long, float, double, boolean, String, or enum)</li>
+	 *         <li>A complex type (a {@link ValueList} or {@link ValueMap} of basic or complex types)</li>
+	 *         <li>An object whose class follows these rules</li>
+	 *     </ul>
+	 * </ul>
+	 *
```

Comment:
It would probably be a good idea to note things that are never serialized, like (i presume) static fields, and fields with the `transient` keyword (which you should implement, by the way)